mllib/src/main/scala/org/apache/spark/ml/feature/RFormulaParser.scala

```diff
@@ -109,7 +157,15 @@ private[ml] object RFormulaParser extends RegexParsers {
   def columnRef: Parser[ColumnRef] =
     "([a-zA-Z]|\\.[a-zA-Z_])[a-zA-Z0-9._]*".r ^^ { case a => ColumnRef(a) }
 
-  def term: Parser[Term] = intercept | columnRef | "\\.".r ^^ { case _ => Dot }
+  def dot: Parser[InteractionComponent] = "\\.".r ^^ { case _ => Dot }
+
+  def interaction: Parser[List[InteractionComponent]] = repsep(columnRef | dot, ":")
```

Comment:
If we want to separate `columnRef` from `interaction`, we should use `rep1sep` here and update `term`. This might add several lines of code, but make the code easier to understand.